### ISSUE TITLE
Unexpected '[' on line 166 of butter.php & Board Links

### ISSUE THREAD

**Issue by WASasquatch** [MitsubaBBS/Mitsuba#71]
I believe it's because there are no posts that it is trying to fetch from, is there a simple work around for this?  Line of code in question is the fetch_array

$result = $this->conn->query("SELECT sum(orig_filesize) FROM posts");
$num_bytes = $result->fetch_array()[0];

**Comment by WASasquatch** [comment]
Here is my fix, works well enough

$result = $result->fetch_array();
if (!isset($result) && !is_array($result)) {
    $num_bytes = 0;
} else {
    $num_bytes = $result[0];
}

And allows me to continue forward with rebuilding cache. However... no I am stuck.

When I make board links, no matter what, they aren't links. Just plain text.

**Comment by pavulon** [comment]
This error is present only in PHP versions lower than 5.4, where you can't do the function array dereferencing.

**Comment by WASasquatch** [comment]
Yeah I found that out. What about board links? They don't ever seem to generate links at all, even with a URL, base root to board, or w/e. Just can't seem to get it to produce a link, it just becomes text. Is this another PHP 5.3 issue as compared to 5.4?

**Comment by pavulon** [comment]
Probably. I can't reproduce the bug on 5.5.

**Comment by WASasquatch** [comment]
Ehh, WAMP needs to upgrade. Looks like I need to manually install pho and
apache.

**Comment by pavulon** [comment]
You can try out XAMPP, it's pretty similar.

**Comment by WASasquatch** [comment]
I think ill just hold off until the dedi is up and I can vm a debian distro
to spare this trouble. Windows is evil
